src/lib/redis/cluster.c

```diff
@@ -1472,6 +1480,31 @@ void *fr_redis_cluster_conn_create(TALLOC_CTX *ctx, void *instance, fr_time_delt
 		return NULL;
 	}
 
+#ifdef HAVE_REDIS_SSL
+	if (node->cluster->conf->ssl) {
+		redisSSLContextError ssl_error = REDIS_SSL_CTX_NONE;
+#ifdef HAVE_REDIS_SSL_OPTIONS
+		redisSSLOptions opt = { .verify_mode = REDIS_SSL_VERIFY_PEER };
+
+		ssl_context = redisCreateSSLContextWithOptions(&opt, &ssl_error);
```

Comment:
The redis SSL functions don't expose many useful options, but as the whole thing is based around OpenSSL it looks like it's mostly just taking an `SSL_CTX *`.

https://github.com/redis/hiredis/blob/master/ssl.c#L64

You can get an `SSL_CTX` using the generic TLS client config code https://github.com/FreeRADIUS/freeradius-server/blob/master/src/lib/tls/conf.c#L495 and another function which transforms that into a configured `SSL_CTX`.

https://github.com/FreeRADIUS/freeradius-server/blob/master/src/lib/tls/ctx.c#L610

Reply:
We would somehow need to get at the `SSL *` to call `fr_tls_session_request_bind` and `fr_tls_session_request_unbind`, but after that all the async cert validation code should work properly too.

Reply:
Nice, looks like we can alloc the `SSL *` too, so we can configure it the same as all the other `SSL *` objects we create: https://github.com/redis/hiredis/blob/master/ssl.c#L395

Reply:
@arr2036 yes, you can merge with changes - I don't mind as long as it works.

On a side note I would not rely on contexts to be the same as they may want to change that "opaque" object later